### ISSUE TITLE
Support Rum

### ISSUE THREAD

**Issue by webmonarch** [drapanjanas/re-natal#37]
I am interested in supporting Rum https://github.com/tonsky/rum for react-native development through re-natal.

Rum is a minimal Clojure wrapper around React that is opinion-less.

I took an initial stab at getting it to work but I was having trouble with sablono https://github.com/r0man/sablono (it seems tightly coupled with browser DOM stuff).  

Shimming a couple things and writing some custom Rum macros and mixins to not use sablono should be all that it takes.  

Putting this out there if anyone else is interested as well and/or has insights.

**Comment by drapanjanas** [comment]
That isnteresting, and should not be hard to do I guess. You can take the re-natal interface 'reagent6' as an example as it also has some shimmed namespaces to overcome coupling with DOM. If you have already working manually built setup with re-natal, I could help to convert it into an `interface` option.

**Comment by webmonarch** [comment]
Thanks @drapanjanas for the response. I have a skeleton I am working on locally. I've gotten everything to compile...still working on the hello world. Once I get something functional end-to-end I'll let you know.  It'll probably be next week sometime as I am attending [Clojure/West](http://clojurewest.org/) this weekend 😎 .  Cheers 🍻

**Comment by webmonarch** [comment]
@drapanjanas I have pushed a branch with initial Rum support.

https://github.com/webmonarch/re-natal/tree/add-rum
https://github.com/webmonarch/re-natal/commit/d931ee1700328533d6fcf355a4eec9273170812a

Learned a few things and was able to leverage a few things from the om.next impl.

There are a couple of last mile things that are somewhat Rum related but also somewhat React Native  related, specifically:

* component reload/re-render on figwheel signal
* component reload/re-render on app-state atom changes

I was trying to parse the way the om.next and reagent templates handle this... 

Thoughts and feedback appreciated.  Cheers!

**Comment by drapanjanas** [comment]
Cool, I will check it out over the weekend maybe I will be able to come up with something. 
But I'll need to learn a bit about Rum. But from the first glance Rum examples are also based on datascript. Would it be a good idea to use datascript also in Rum tremplate? I do not know what are "natural" choices with rum.

**Comment by webmonarch** [comment]
Thanks @drapanjanas. I've been marinating on it more. The main block for me right now is the swapping the `AppRoot` component on fighweel signal.  Since the root React Native component is a singleton, it's not just as easy as re mounting the component into the DOM.  

I **think** we'll need a minimal React component that contains/wraps the `AppRoot` component.  I think it's similar to what the om-next example is doing...but I need to parse the code a bit more.  Once we have the root element reloading working, re-rendering on state atom change is Rum specific and I can do that.

**Comment by webmonarch** [comment]
So @drapanjanas I finally got my head around what I wanted to do for figwheel reloading.  It's inspired by the `om-next.support` namespace:  https://github.com/drapanjanas/re-natal/pull/45/files#diff-b5e095a40f0bc729e256dba31b8a9689R1

Specifically, it creates a singleton root react component that supports mounting/remounting of the changed app component.  I actually like this solution better than the om templates as it is more general and has nothing to do with Rum or other React frameworks.  But to be fair, om's is more complicated because om is more complicated. 😄 

Closing this issue in favor of the PR #45
